src/client/pfs.go

```diff
@@ -447,6 +447,10 @@ func (c APIClient) PutFileWriter(repoName string, commitID string, path string,
 
 // PutFile writes a file to PFS from a reader.
 func (c APIClient) PutFile(repoName string, commitID string, path string, reader io.Reader) (_ int, retErr error) {
+	c.streamSemaphores <- struct{}{}
+	defer func() {
+		<-c.streamSemaphores
+	}()
```

Comment:
Minor stylistic thing, I generally like to make these into 1 liners to conserve vertical space.
Gofmt doesn't mind it either.